File: PHP/Film.php
<?php
class Film {
    private $id_film;
    private $judul;
    private $genre;
    private $sutradara;
    private $foto; // Atribut untuk path gambar lokal

    public function __construct($id, $jud, $gen, $sut, $foto) {
        $this->id_film = $id;
        $this->judul = $jud;
        $this->genre = $gen;
        $this->sutradara = $sut;
        $this->foto = $foto;
    }

    public function getId() { return $this->id_film; }
    public function getJudul() { return $this->judul; }
    public function getGenre() { return $this->genre; }
    public function getSutradara() { return $this->sutradara; }
    public function getFoto() { return $this->foto; }

    // Tambahan Setter untuk ID agar bisa di-update
    public function setId($id) { $this->id_film = $id; }
    
    public function setJudul($jud) { $this->judul = $jud; }
    public function setGenre($gen) { $this->genre = $gen; }
    public function setSutradara($sut) { $this->sutradara = $sut; }
    public function setFoto($foto) { $this->foto = $foto; }
}
?>

File: Python/Film.py
class Film:
    def __init__(self, id_film, judul, genre, sutradara):
        self.__id_film = id_film
        self.__judul = judul
        self.__genre = genre
        self.__sutradara = sutradara

    # --- GETTERS ---
    def get_id(self): 
        return self.__id_film
        
    def get_judul(self): 
        return self.__judul
        
    def get_genre(self): 
        return self.__genre
        
    def get_sutradara(self): 
        return self.__sutradara

    # --- SETTERS ---
    # Ini fungsi yang tadi bikin error karena belum ada
    def set_id(self, id_film): 
        self.__id_film = id_film
        
    def set_judul(self, judul): 
        self.__judul = judul
        
    def set_genre(self, genre): 
        self.__genre = genre
        
    def set_sutradara(self, sutradara): 
        self.__sutradara = sutradara